lib/instrumentation/trace-context.js

```diff
@@ -107,23 +107,13 @@ const bufferSymbol = Symbol('trace-context-buffer')
 class TraceContext {
   constructor (buffer) {
     this[bufferSymbol] = buffer
+    this.recorded = !!(buffer[OFFSETS.flags] & FLAGS.recorded)
```

Comment:
Should we continue using `Object.defineProperty(...)` to keep this immutable?

```suggestion
    Object.defineProperty(this, 'recorded', {
      value: !!(buffer[OFFSETS.flags] & FLAGS.recorded),
      enumerable: true
    })
```